enterprise/server/cmd/goinit/BUILD

```diff
@@ -25,6 +25,14 @@ go_library(
 go_binary(
     name = "goinit",
     embed = [":goinit_lib"],
-    pure = "on",
-    static = "on",
+    # "pure" causes the init file to require an ELF interpreter which the kernel
+    # does not have access to if we are bulding statically.
+    pure = select({
+        "//:static": "auto",
+        "//conditions:default": "on",
+    }),
```

Comment:
The problem shows up if you use `--@io_bazel_rules_go//go/config:static=true`, in my experience. Not sure why we'd make a test to confirm that a build state we don't use doesn't work.

Reply:
Hmm, I was confused which line was which when I made the comment above.

I was thinking that we could have a test to help confirm that our binary was statically linked, like using `ldd <binary>` to verify the binary as we discussed in Slack.
This will help us prevent future regressions where our binary could accidentally become dynamically linked 🤔 .

It's probably just a "nice to have" nit and should not block this PR though. Please feel free to ignore.